IntegrationWithPharmacies/Controllers/ActionsAndBenefitsController.cs

```diff
@@ -9,20 +12,37 @@ namespace IntegrationWithPharmacies.Controllers
     [ApiController]
     public class ActionsAndBenefitsController : Controller
     {
-            private MessageService MessageService { get; set; }
-            public ActionsAndBenefitsController(MyDbContext context)
-            {
-                 MessageService = new MessageService(context);
-            }
+        private MessageService MessageService { get; set; }
+        private RegistrationInPharmacyService RegistrationInPharmacyService { get; set; }
+   
+        public ActionsAndBenefitsController(MyDbContext context)
+        {
+            MessageService = new MessageService(context);
+            RegistrationInPharmacyService = new RegistrationInPharmacyService(context);
+        }
+
+        [HttpGet]
+        public IActionResult Get()
+        {
+           return Ok(getAllForVue());
+        }
 
-            [HttpGet]
-            public IActionResult Get()
+        private List<Message> getAllForVue()
+        {
+            List<Message> messageForVue = new List<Message>();
+            foreach(Message message in MessageService.GetAll())
             {
-                return Ok(MessageService.GetAll());
+                RegistrationInPharmacy registration = RegistrationInPharmacyService.getPharmacyName(GetName(message));
+                if (registration != null) { messageForVue.Add(message); }
             }
+            return messageForVue;

```

Comment:
-Refactor code. Replace method getAllForVue() with Where syntax, for example :
return MessageService.GetAll().Where(message => (RegistrationInPharmacyService.getPharmacyName(GetName(message))!=null) )
-Add a method that checks if the object is null and apply it in refactoring.